modules/instrument_builder/js/instrument_builder.instrument.js

```diff
@@ -357,9 +358,226 @@ var Instrument = {
+                    break;
+                  case "text":
+                    // Date element type
+                    if (validationType == "date_dmy"
+                      || validationType == "date_ymd"
+                    ) {
+                      let dropdown = "";
+                      let dateFormat = "";
+                      // If element name ends with "_date", make it a standard date
+                      // Otherwise, it can be a basic date
+                      if (elementName.substring(elementName.length - 5) == "_date") {
+                        // provide not_answered option
+                        dropdown = "select{@}" + elementName + "_status" +
+                          "{@}{@}NULL=>''{-}'not_answered'=>'Not Answered'\n";
+                      } else {
+                        dateFormat = "BasicDate";
+                      }  
+                      lineContent += "date{@}";
+                      lineContent += elementName + "{@}" + elementDescription;
+                      lineContent += "{@}" + minValue;
+                      lineContent += "{@}" + maxValue;
+                      lineContent += "{@}" + dateFormat + "\n";
+                      lineContent += dropdown;
+                    } else if (validationType == "number") {
+                    // Numeric element type
+                      lineContent += "numeric{@}";
+                      lineContent += elementName + "{@}" + elementDescription;
+                      lineContent += "{@}" + minValue;
+                      lineContent += "{@}" + maxValue + "\n";
+                      lineContent += "select{@}" + elementName + "_status" +
+                        "{@}{@}NULL=>''{-}'not_answered'=>'Not Answered'\n";
+                    } else {
+                    // Text
+                      lineContent += "text{@}";
+                      lineContent += elementName + "{@}" + elementDescription + "\n";
+                      lineContent += "select{@}" + elementName + "_status" +
+                        "{@}{@}NULL=>''{-}'not_answered'=>'Not Answered'\n";
+                    }
+                    break;
+                  case "notes":
+                    lineContent += "textarea{@}";
+                    lineContent += elementName + "{@}" + elementDescription + "\n";
+                    lineContent += "select{@}" + elementName + "_status" +
+                      "{@}{@}NULL=>''{-}'not_answered'=>'Not Answered'\n";
+                    break;
+                  case "calc":
+                    lineContent += "static{@}";
+                    lineContent += elementName + "{@}" + elementDescription + "\n";
+                    elementName = '';
+                    break;
+                  case "descriptive":
+                    lineContent += "static{@}{@}" + elementDescription + "\n";
+                    elementName = '';
+                    break;
+                }
+                // Add rules
```

Comment:
added ability to read branching logic in csv and download .rules file along side .linst file